### DISCUSSION TITLE
Segment priority: UI issue

### DISCUSSION THREAD

**Discussion by adekbadek** [Automattic/newspack-popups#816] [Ideas]
_As it turns out, it's a feature not a bug - the issue is with the UI, which can be misleading._

---

## Steps to reproduce

1. Create two segments which would match a first-time user, and two overlay prompts 
2. Assign one prompt to a segment, and the other one to "Everyone" segment
2. Set the prompt-less segment to highest priority
3. Visit the site as a non-logged-in user, observe the segment-less prompt is displayed instead of the segmented one

**Comment by dkoo** [comment]
Isn't this the expected behavior, though, due to non-cascading segment matching? Given:

* Segment A (highest priority): Has no prompts
* Segment B (second priority): Has Overlay Prompt 1
* Everyone: Has Overlay Prompt 2

As a reader, I match both Segments A and B. Because Segment A has no prompts, I'm not shown any prompts belonging to segments with a lower priority than Segment A, _unless_ they belong to "Everyone". Hence, I see Overlay Prompt 2.

**Comment by adekbadek** [comment]
Hm, I've understood segments as always having a higher priority than "Everyone". Assuming I only have overlay prompts (not a good practice, but to illustrate a point) and the following setup:
1. "Not subscribed" segment, newsletter signup form in prompt
2. "Not donated" segment, donation form in prompt
3. Everyone, article recommendation in prompt

I'd like to maximise subscribers, then donors, and only if someone is already a subscriber and a donor (or has dismissed other prompts), display the generic "house ad". 
Now, if for some reason there's no prompt assigned to my top segment ("Not subscribed"), then I'd expect the next-priority segment messaging to kick in ("Not donated"). That's what I image looking at the UI, at least - the "Everyone" segments is there at the bottom of segments list (not a separate view).

**Comment by adekbadek** [comment]
Turns out I was advocating for the old, cascading segments approach (see #445).
Maybe it's just me, but the UI suggests it's a cascade - ending with the "Everyone" segment. @thomasguillot maybe we could separate the "Everyone" section, so that it's more understandable that a reader would match one of the segments _and_ then the "Everyone" prompts. The UI suggests that "Everyone" is one of the segments, while really it's a different feature - targeting readers regardless of segment (which is written as the description, but the layout does not suggest that).

**Comment by dkoo** [comment]
I definitely agree that the current UI suggests the old "cascading" behavior of segmentation and would advocate for a ground-up rethinking of the UI in both Campaigns and Segments tabs given the current behavior.
